### PR TITLE
feat(alearm246): create isOwner function

### PR DESCRIPTION
**Pull request by alearm246**

### CODE REVIEW

File: utils/isOwner.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  try {
+    const singleCollection = await collection.findOne({ id });
+
+    if(userId === singleCollection.user.id){
+      return true;
+    }
+
+    return false;
+  } catch(err) {
+    console.error(err);
```
Comment (chungthuang) [inline utils/isOwner.js:11]: What will this function return in this case?

File: production/kustomization.yaml [hunk 2]
```diff
@@ -1,10 +0,0 @@
-apiVersion: kustomize.config.k8s.io/v1beta1
```
Comment (chungthuang) [inline production/kustomization.yaml:1]: Why did you delete this file and configmap?

Comment (alearm246) [reply production/kustomization.yaml:1]: That's weird because I don't remember deleting that

File: utils/isOwner.js (hunk 1 above)
Comment (alearm246) [reply utils/isOwner.js:11]: Yeah I think it should return false

File: utils/isOwner.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+const isOwner = async (collection, userId, id) => {
+  try {
+    const singleCollection = await collection.findOne({ id });
+
+    if(userId === singleCollection.user.id){
```
Comment (chungthuang) [inline utils/isOwner.js:5]: What if record can't be found? I think we should check singleCollection is not null first. Beyond that, does ever collection have a user.id field?

Comment (alearm246) [reply utils/isOwner.js:5]: That's true I forgot to check  if the record can't be found.  There are some collections that don't have a user id field, but I figured this function would only be used on collections that do.

Comment (chungthuang) [reply utils/isOwner.js:5]: Is the try catch going to prevent crashing the app if the collection doesn't have user id field?

Comment (alearm246) [reply utils/isOwner.js:5]: That I have not tested. Do you think I should add a check that checks if the collection doesn't have a user id? My thought process was that this function will only ever be used on collections that do have user id fields, but if you think I should add it I'll add it.

Comment (chungthuang) [reply utils/isOwner.js:5]: It's not always safe to assume that the caller will pass in good inputs.

Comment (alearm246) [reply utils/isOwner.js:5]: got it

File: utils/isOwner.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    if(!singleCollection) throw new Error('A record with that id was not found');
+
+    if(!singleCollection.user?.id) throw new Error('This collection does not have a user id');
+
+    if(userId === singleCollection.user.id){
+      return true;
+    }
+
+    return false;
+  } catch(err) {
```
Comment (chungthuang) [inline utils/isOwner.js:14]: The error you throw in the try branch will be caught here. Just want to make sure this is what you intended.

Comment (alearm246) [reply utils/isOwner.js:14]: Ah I thought it would be caught in the catch block where I called the function

Comment (alearm246) [reply utils/isOwner.js:14]: My original idea was that those errors would flow on top and be caught in the catch block and then I could return a HTTP error message back to the caller of the API

Comment (alearm246) [reply utils/isOwner.js:14]: Did some googling and figured out I could bubble up the error if I rethrow the exception of the catch block in the isOwner function. The only thing is I won't be able to return false since throw exists the function.

Comment (chungthuang) [reply utils/isOwner.js:14]: I think it's better to return false and let the caller decides the HTTP error. There is a chance that the caller doesn't catch the error if this function returns an error.

**Review by chungthuang — APPROVED** [review]

File: utils/isOwner.js (hunk 4 above)
Comment (alearm246) [reply utils/isOwner.js:14]: so for the two checks on lines 5 and 7 I should return false instead? How would they differentiate from the error?

File: utils/isOwner.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+    if(!singleCollection.user?.id) return strapi.errors.badRequest('This collection does not have a user id');
+
+    if(userId === singleCollection.user.id){
+      return true;
+    }
+
+    return false;
+  } catch(err) {
+    strapi.errors.badRequest(err);
```
Comment (chungthuang) [inline utils/isOwner.js:15]: You already catch bad user inputs in the try branch. I think errors that reach the catch branch are internal errors, so semantically it's better to return status 500 here.
